### ISSUE TITLE
Geometries after object query stay present even after closing details

### ISSUE THREAD

**Issue by dktue** [openstreetmap/openstreetmap-website#2082]
When you click "object query" (Objektabfrage) from the context-menu on the map and you close the element-details afterwards, the geometries stay present on the map until you reload.

**Comment by tomhughes** [comment]
I can't reproduce this, mostly because I don't understand how you are closing the details with the geometries displayed.

What should happen is that the side bar appears and as you hover over entries the geometries are shown - when you go to click on the `X` icon you move away from the items and the geometry you were over is removed before you can actually get to the `X` to close it.

What exactly is happening for you? How are you able to close the panel with the geometries displayed?

**Comment by dktue** [comment]
After querying for this building:
https://www.openstreetmap.org/#map=19/48.51477/9.06117
and closing the details-pane on the left, the geometries stay present on the map:
![image](https://user-images.githubusercontent.com/2878623/49806120-2cc6ac80-fd57-11e8-8ffc-6a6cff3d7609.png)

**Comment by tomhughes** [comment]
Yes that show the result you claim to have achieved but not how you got there.

I saw again, how did you move your mouse from hovering over that building in the results (which is what causes the geometry to appear) to the `X` icon to close the sidebar without the geometry disappearing as it does for me?

Or did you click on the entry in the sidebar? That would make the geometry persistent, but it should still go away when the sidebar is closed, or at least it does for me.

**Comment by dktue** [comment]
Here comes the screen-recording: [geometries-stay-present-after-close.zip](https://github.com/openstreetmap/openstreetmap-website/files/2667913/geometries-stay-present-after-close.zip)

**Comment by tomhughes** [comment]
Right, so you are clicking through to the object details.

I can't reproduce that though - when I do that and then close the details the geometry goes away.

What browser are you using? Are there any errors in the javascript console?

**Comment by tomhughes** [comment]
On firefox, yes, with that exact building. I haven't tried it on chrome yet.

**Comment by tomhughes** [comment]
Can't reproduce it on chromium either.

I did just realise one thing though - the geometry on the browse page has a fill while the geometry on the search results doesn't. When you click the `X` in your video the fill is disappearing meaning that it has apparently got both geometries rendered and has removed one but not the other.

**Comment by dktue** [comment]
Yes, that definitely is the case. Only the outlines stay there:

![image](https://user-images.githubusercontent.com/2878623/49808978-0e17e400-fd5e-11e8-8851-629f0e542f94.png)

**Comment by dktue** [comment]
I just wanted to update this: Problem still occurs with Chrome 75.0.3770.142.

**Comment by danieldegroot2** [comment]
First time seeing it myself today with Chrome 151.0.7922.76 on Windows 10

1. Click query features
2. Click on a location to query features
3. Select a feature
4. After it is done loading details, close the details pane using the "X" button.
5. Highlight stays stuck as in https://github.com/openstreetmap/openstreetmap-website/issues/2082#issuecomment-446215994
6. Click on a location again
7. Select a different feature
8. After it is done loading details, close the details, there is now two stuck highlights as in https://github.com/openstreetmap/openstreetmap-website/issues/2082#issuecomment-446233263

Related #5838 in which this did not happen when sidebar is closed after details are done loading.
indicating this may have been previously resolved but occurs again due to recent changes.

Unrelated but similar #3465 in which accumulating highlights appeared in the data layer.
As a sidenote, the data layer still works okay.

**Comment by pablobm** [comment]
I can reproduce with Google Chrome, Version 151.0.7922.76 (Official Build) (arm64), on macOS 15.7.7 (24G720).

**Comment by 1ec5** [comment]
A lot of the highlighting functionality is implemented in [leaflet-osm](https://github.com/openstreetmap/leaflet-osm/), which we’re replacing as part of #6234. We’ll still need to synchronize the highlights with the state of the sidebar.
